### DISCUSSION TITLE
DBeaver postgresql uses local timezone and not Server timezone

### DISCUSSION THREAD

**Discussion by sikevspo** [dbeaver/dbeaver#14432] [Q&A] [answered]
Hi Team, I am running a function on dbeaver and had an error with the date time format of my migration. When I do check SELECT NOW(), instead of showing my server timezone (should be UTC+1) it shows my local pc timezone (UTC+8)

Here is the screen shot:
![image](https://user-images.githubusercontent.com/87894715/139386677-ca016fa4-5f7f-484b-b601-c7acdeb49c47.png)

But when I checked pgAdmin it shows the correct timezone:
![image](https://user-images.githubusercontent.com/87894715/139386767-7cfe66fb-f693-49dd-8600-d03e1ce82f8c.png)

I saw an answer on some discussions that modifying the timezone on DBeaver.ini will change the timezone, it works but I handle multiple servers with multiple timezone and it's not the best solution to modify the config every time.

I also tweak and playaround the date format on connection settings but it does not work.
![image](https://user-images.githubusercontent.com/87894715/139387264-243fb140-3a89-4222-bfa9-564ebe9d0603.png)

Is there somebody experiencing this issue? Do you have any workaround? Thanks!

Here is my DBeaver version:
![image](https://user-images.githubusercontent.com/87894715/139387558-5eeaa955-b87f-4b3c-8a49-09c0b0d40fb8.png)

**Comment by LonwoLonwo** [comment]
Hello @sikevspo 

Please try to disable the "Disable sate/time formatting" button in Data Formats in the Preferences.
Also, you can set Timezone for data source manually on the Data Formats page.

![2021-11-01 10_15_32-Preferences](https://user-images.githubusercontent.com/45152336/139635674-975e4cd1-be48-4132-bcaf-fcffdec99775.png)

**Reply by sikevspo** [reply]
Hi @LonwoLonwo Thank you for your response! I tried changing it on datasource settings but I am having the same result.
![image](https://user-images.githubusercontent.com/87894715/139784959-ef3f8f48-c17b-4a2a-8220-dc7b6a1d30f2.png)


I already change timezone to UTC, GMT, GMT + 1:00, UTC+1, +1. But it is still the same.

![image](https://user-images.githubusercontent.com/87894715/139785027-9e568c13-13bf-4b31-b433-f76953819820.png)

I am also looking a way to add a driver property to set timezone to server timezone or manually create a JDBC url to be used, but there are no good references I could use. Any idea with those?

Thanks~

**Reply by LonwoLonwo** [reply]
And what about 

> Please try to disable the "Disable sate/time formatting" button in Data Formats in the Preferences.

Didn't help also?

**Reply by sikevspo** [reply]
No it did not help too. I found an alternative solutions on creating bootstrap SQL through connection will post in a while. Thanks for the help! @LonwoLonwo

**Comment by sikevspo** [comment, marked as answer]
Hi,
I found an alternative solution but this only works for postgres for now, but I think there is an equivalent script for other database program.

1. If you have existing connection, go to edit connection then connection settings: ( will not include db details due to data privacy)
![image](https://user-images.githubusercontent.com/87894715/140018748-ad6aaaa5-25e3-4ef4-8b6a-00e136c174a7.png)

2. Collapse connection settings, then proceed to initialization:
![image](https://user-images.githubusercontent.com/87894715/140018875-95b542ab-04ad-4ac8-897e-f530cde7013e.png)

3. On bootstrap queries property click configure.
![image](https://user-images.githubusercontent.com/87894715/140018927-b0263da0-8937-4930-9a52-ca9a80dffee9.png)

4. On the bootsrap query dialog box click add and enter the following query:

SET TIME ZONE '_preferred timezone_'

In this example I have used 'Europe/Lisbon' timezone.

You may also use 
SET timezone TO '_preferred  timezone_' 
command. For more info see postgres set time zone documentation.

![image](https://user-images.githubusercontent.com/87894715/140019132-f533993e-ba28-4d8a-84e2-fc60dd3bc16f.png)

5. Reconnect the DB and try to select NOW(); it should work.

Note that this is only alternative solution, I was hoping DBeaver team can add it as a feature to modify timezone and if possible much preferably it will set default timezone as server time zone. Date time data is very critical for us data developers and analyst.

More power to the team. Thanks!

**Reply by LonwoLonwo** [reply]
Interesting idea. Thanks!
